src/test/java/org/example/servletsHomework/servlet/functionalTests/FunctionalTextServletTest.java

```diff
@@ -0,0 +1,337 @@
+
+        String expectedOutputDeleteAll = "All text has been deleted";
+        assertEquals(expectedOutputDeleteAll, responseDeleteAll.body());
+        assertEquals(HttpServletResponse.SC_OK, responseDeleteAll.statusCode());
+
+        HttpRequest requestGetAll = HttpRequest.newBuilder()
+                .uri(new URI(baseUrl + textUrl))
+                .GET()
+                .header("token", token)
+                .build();
+
+        HttpResponse<String> response = client.send(requestGetAll, HttpResponse.BodyHandlers.ofString());
+
+        String expectedOutputGetAll = "[]";
+        assertEquals(expectedOutputGetAll, response.body());
+        assertEquals(HttpServletResponse.SC_OK, response.statusCode());
+    }
+
+
+    @Test
+    void shouldPostTextThenGetAllTexts() throws URISyntaxException, IOException, InterruptedException {
+
+        String jsonRequest = """
+                {
+                "text": "TestText123"
+                }
+                """;
+
+        List<String> expectedId = new ArrayList<>();
+
+        for (int i = 0; i < 3; i++) {
+            HttpRequest requestPost = HttpRequest.newBuilder()
+                    .uri(new URI(baseUrl + textUrl))
+                    .POST(HttpRequest.BodyPublishers.ofString(jsonRequest))
+                    .header("token", token)
+                    .header("Content-Type", "application/json")
+                    .build();
+
+            HttpResponse<String> responsePost = client.send(requestPost, HttpResponse.BodyHandlers.ofString());
+            assertEquals(HttpServletResponse.SC_CREATED, responsePost.statusCode());
+
+            String responseBody = responsePost.body();
+            assertTrue(responseBody.contains("\"text\":\"TestText123\""));
+            assertTrue(responseBody.contains("\"message\":\"Text saved with id:"));
+
+            String messageStart = "\"textId\":";
+            int idStartIndex = responseBody.indexOf(messageStart) + messageStart.length();
+            int idEndIndex = responseBody.indexOf("}", idStartIndex);
+            String textId = responseBody.substring(idStartIndex, idEndIndex).trim();
+
+            expectedId.add(textId);
+        }
+
+        HttpRequest requestGetAll = HttpRequest.newBuilder()
+                .uri(new URI(baseUrl + textUrl))
+                .GET()
+                .header("token", token)
+                .build();
+
+        HttpResponse<String> responseGet = client.send(requestGetAll, HttpResponse.BodyHandlers.ofString());
+
+        assertEquals(HttpServletResponse.SC_OK, responseGet.statusCode());
+
+        String responseBody = responseGet.body();
+
+        for (String id : expectedId) {
+            assertTrue(responseBody.contains("textId=" + id));
+            assertTrue(responseBody.contains("value='TestText123'"));
+        }
+    }
+
+    @Test
+    void shouldRedirectToLoginWhenTokenExpires() throws Exception {
+
+        Thread.sleep(185000);
```

Comment:
What will be with your test when requirements changed to expire token in 2 hours ?